lib/sinatra/respond_with.rb

```diff
@@ -1,7 +1,7 @@
 require 'sinatra/json'
 require 'sinatra/base'
 
-$KCODE = "UTF-8"
+$KCODE = "UTF-8" unless RUBY_VERSION > "1.9.0"
```

Comment:
This works, also we don't support 1.9.0, so it's probably fine.

Alternatively we could do: `RUBY_VERSION.to_f >= 1.9`